tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,104 @@
+import copy
+import pytest
+
+from configuration import available_ports, ALL_TEST_CURVES, MINIMAL_TEST_CERTS
+from common import Ciphers, ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version, to_bytes
+
+PADDING_SIZE_MIN = 0
+PADDING_SIZE_SMALL = 250
+PADDING_SIZE_MEDIUM = 1000
+PADDING_SIZE_MAX = 1 << 14
+
+PADDING_SIZES = [
+    PADDING_SIZE_MIN,
+    PADDING_SIZE_SMALL,
+    PADDING_SIZE_MEDIUM,
+    PADDING_SIZE_MAX
+]
+
+PAYLOAD_SIZE_SMALL = 10
+PAYLOAD_SIZE_MEDIUM = 100
+# openssl has a max fragment length of 4096 bytes
+PAYLOAD_SIZE_LARGE = 4096
+
+OPENSSL_PAYLOAD_SIZES = [PAYLOAD_SIZE_SMALL,
+                         PAYLOAD_SIZE_MEDIUM, PAYLOAD_SIZE_LARGE]
+
+# https://www.rfc-editor.org/rfc/rfc5116
+TLS13_CIPHERS_WITH_16_BYTE_TAGS = [
```

Comment:
Why is it important that we only test ciphers with 16-byte tags?

Reply:
It's to make predicting the payload size easier. The test checks the openssl output and validates that the payload size is expected (and therefore padded as  expected). If the test specifies openssl to pad to `n` bytes then the total payload size is then `n + 16` bytes since we know we're using 16-byte aead tags

Reply:
What happens when we use the other TLS1.3 cipher. Do we not know how large the record will be?

Reply:
The other tls 1.3 cipher being chacha20? Looks like it's also produces a 16 byte tag. I can add it as well.